app/workers/judge_submission_worker.rb

```diff
@@ -68,14 +68,34 @@ def judge
     result = {}
     setup_judging do
       if submission.language.compiled
-        result['compile'] = compile!(exe_filename) # possible caching
-        return result.merge!(grade_compile_error(result['compile'])) if result['compile']['stat'] != 0 #error
+        result['compile'] = compile!(submission.source, submission.language, exe_filename) # possible caching
+        return result.merge!(grade_compile_error(result['compile'])) if result['compile']['stat'] != 0 # error
       else
         File.open(File.expand_path(exe_filename, tmpdir),"w") { |f| f.write(submission.source) }
       end
 
       run_command = submission.language.run_command(exe_filename)
 
+      if problem.evaluator.nil?
+        eval_command = nil
+      else
+        if problem.evaluator.language.nil?
+          eval_command = "./#{EvalFileName}"
+        else
+          eval_command = problem.evaluator.language.run_command(EvalFileName)
+        end
+
+        if problem.evaluator.language&.compiled
+          evaluator_compilation = compile!(problem.evaluator.source, problem.evaluator.language, EvalFileName) # possible caching
+          raise evaluator_compilation['log'] if evaluator_compilation['stat'] != 0 # error
```

Comment:
It seems a little strange to raise an error if evaluator compilation fails. This leaves submissions pending and doesn't make it clear that evaluator compilation failed and why it failed (without looking on qless). Also, people with the staff role have the permission to create evaluators but wouldn't be able to see their evaluator's compilation errors.

I imagine it's sensible to set the status to the submission as errored (2) when this occurs instead of raising an error and then store + display the evaluator compilation log to staff/admin viewing the submission.

Reply:
Thanks for reviewing. I think this is a reasonable suggestion, and have pushed some changes to address this.